internal/state/graph/httproute.go

```diff
@@ -666,3 +682,71 @@ func validateFilter(
 
 	return allErrs
 }
+
+func validateFilterHeaderModifier(
+	validator validation.HTTPFieldsValidator,
+	filter v1beta1.HTTPRouteFilter,
+	filterPath *field.Path,
+) field.ErrorList {
+	headerModifier := filter.RequestHeaderModifier
+
+	headerModifierPath := filterPath.Child("requestHeaderModifier")
+
+	if headerModifier == nil {
+		panicForBrokenWebhookAssumption(errors.New("requestHeaderModifier cannot be nil"))
+	}
+
+	return validateFilterHeaderModifierFields(validator, headerModifier, headerModifierPath)
+}
+
+const multipleHeaderFieldsErr = "header names in requestHeaderModifier cannot be specified more than once"
+
+func validateFilterHeaderModifierFields(
+	validator validation.HTTPFieldsValidator,
+	headerModifier *v1beta1.HTTPHeaderFilter,
+	headerModifierPath *field.Path,
+) field.ErrorList {
+	var allErrs field.ErrorList
+	uniqueHeaderNames := make(map[string]struct{})
+
+	for _, h := range headerModifier.Add {
+		if _, ok := uniqueHeaderNames[string(h.Name)]; ok {
```

Comment:
@pleshakov do we need to perform this validation for a third time? I think it makes sense with nil values because they will panic, but it seems like unnecessary work for this type of validation.

Reply:
@kate-osborn  this will not lead to the control plane or NGINX reload failure , so I think we can avoid validating it.